roles/lib_utils/src/generate.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python
+'''
+  Generate the openshift-ansible/roles/lib_openshift_cli/library/ modules.
+'''
+
+import os
+
+# pylint: disable=anomalous-backslash-in-string
+GEN_STR = "#!/usr/bin/env python\n"                                  + \
+          "# pylint: disable=missing-docstring\n"                    + \
+          "#     ___ ___ _  _ ___ ___    _ _____ ___ ___\n"          + \
+          "#    / __| __| \| | __| _ \  /_\_   _| __|   \\\n"        + \
+          "#   | (_ | _|| .` | _||   / / _ \| | | _|| |) |\n"        + \
+          "#    \___|___|_|\_|___|_|_\/_/_\_\_|_|___|___/_ _____\n"  + \
+          "#   |   \ / _ \  | \| |/ _ \_   _| | __|   \_ _|_   _|\n" + \
+          "#   | |) | (_) | | .` | (_) || |   | _|| |) | |  | |\n"   + \
+          "#   |___/ \___/  |_|\_|\___/ |_|   |___|___/___| |_|\n"
+
+OPENSHIFT_ANSIBLE_PATH = os.path.dirname(os.path.realpath(__file__))
+
+FILES = {'yedit.py':  ['doc/license', 'doc/yedit', 'class/base.py', 'class/yedit.py', 'ansible/yedit.py'],
```

Comment:
re-commenting because I somehow commented on an out-dated version before...

----

This here, respectfully, no no no.

Let's move the **data** that the generator code uses to build artifacts out of the **code**. The way I see it, this should be a simple data (yaml) file that the generator implicitly defaults to reading, or is provided as an option or other form of input (Most likely just defaulting to reading the source data)

For example: https://gist.github.com/tbielawa/abc2dde72cc51eb2f1c6ff01b6c780b0#file-generator-sources-yaml

Another thought about this, some of those file selections in the array could be implicitly determined from the to-generate-file name if you define a specific pattern for doing so. That is to say, if your `{'yedit.py'}` entry is basically empty then all the default files are automatically included.

Ex: You're generating 'yedit.py' with no cat-files included in the list, then the license file is included automatically, if a 'doc/yedit' named doc file exists then it is included. The base.py is included automanually. The 'src/yedit.py' and ansible/yedit.py' files are included automatically.

You would basically have an easy-mode for generating new artifacts that meet a standard concatenation pattern. See gist below for example:

https://gist.github.com/tbielawa/abc2dde72cc51eb2f1c6ff01b6c780b0#file-easy-mode-generator-yaml